app/renderer/lib/appium-client.js

```diff
@@ -0,0 +1,341 @@
+    } catch (e) {
+      contextsError = e;
+    }
+
+    if (currentContext !== NATIVE_APP) {
+      await this.driver.switchContext(currentContext);
+    }
+
+    /**
+     * If its a webview then update the HTML with the element location
+     * so the source can be used in the native inspector
+     */
+    try {
+      if (currentContext !== NATIVE_APP) {
+        const webviewStatusAddressBarHeight = await this.driver.executeScript(getWebviewStatusAddressBarHeight,
+          [{platformName, statBarHeight}]);
+        await this.driver.executeScript(setHtmlElementAttributes, [{
+          platformName,
+          webviewStatusAddressBarHeight
+        }]);
+      }
+    } catch (ign) {}
+
+    return {contexts, contextsError, currentContext, currentContextError};
+  }
+
+  async getSourceUpdate () {
+    try {
+      const source = parseSource(await this.driver.getPageSource());
+      return {source};
+    } catch (err) {
+      return {sourceError: err};
+    }
+  }
+
+  async getScreenshotUpdate () {
+    try {
+      const screenshot = await this.driver.takeScreenshot();
+      return {screenshot};
+    } catch (err) {
+      return {screenshotError: err};
+    }
+  }
+
+  /**
+   * If the app under test can return contexts command.
+   *
+   * @returns {boolean} True if the app under test supports contexts command.
+   *
+   */
+  async hasContextsCommand () {
+    try {
+      await this.driver.getContexts();
+      return true;
+    } catch (ign) { }
+
+    // If the app under test returns non JSON format response
+    return false;
+  }
+
+  /**
+   * Custom implementation of the `mobile:getContexts` for Android, which only returns an object with the id and title
+   */
+  async getAndroidContexts () {
+    let newContexts = [];
+    const currentContext = await this.driver.getContext();
+
+    // Get the contexts and retrieve extra data
+    const contexts = await this.driver.getContexts();
+    const getContextData = async (context) => {
+      let title;
+      const id = context;
+      if (id !== NATIVE_APP) {
+        await this.driver.switchContext(context);
+        const pageTitle = await this.driver.getTitle();
+        title = _.isEmpty(pageTitle) ? 'No page title available' : pageTitle;
+      }
+      return {
+        id,
+        ...(title ? {title} : {}),
+      };
+    };
+    // const newContexts = await Promise.all(contexts.map((context) => getContextData(context)));
+    for (const context of contexts) {
+      newContexts.push(await getContextData(context));
+    }
+
+    // Set it back to the current context
+    await this.driver.switchContext(currentContext);
+
+    return newContexts;
+  }
+
+  /**
+   * Retrieve available contexts, along with the title associated with each webview
+   */
+  async getContexts (platform) {
+    return _.toLower(platform) === 'ios'
+      ? await this.driver.executeScript('mobile:getContexts', [])
+      : await this.getAndroidContexts();
```

Comment:
`mobile:getContexts` has also been recently added to Android

Reply:
Oh great!